datadog/api/api_client.py

```diff
@@ -0,0 +1,213 @@
+# stdlib
+import logging
+import time
+
+# datadog
+from datadog.api import _api_version, _max_timeouts, _backoff_period
+from datadog.api.exceptions import ClientError, ApiError, HttpBackoff, \
+    HttpTimeout, ApiNotInitialized
+from datadog.api.http_client import RequestClient
+from datadog.util.compat import json, is_p3k
+
+
+log = logging.getLogger('dd.datadogpy')
+
+
+class APIClient(object):
+    """
+    """
+    # HTTP transport parameters
+    _backoff_period = _backoff_period
+    _max_timeouts = _max_timeouts
+    _backoff_timestamp = None
+    _timeout_counter = 0
+    _api_version = _api_version
+
+    def __init__(self):
+        """
+        Instantiate a HTTP Client.
+        """
+        self._http_client = self._get_http_client()
+
+    @staticmethod
+    def _get_http_client():
```

Comment:
I actually don't think this logic should be here; it would be better in `datadog.api.http_client`. This way you can import `from datadog.api.http_client import get_http_client`, and I think it makes more sense to have the logic close to the declaration of the classes. What do you think ?